tools/snapshot/snapshot.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"flag"
+	"log"
+
+	"xrplf/clio/clio_snapshot/internal/export"
+	"xrplf/clio/clio_snapshot/internal/server"
+)
+
+func validateExportFlags(seq, endSeq uint, path, grpcServer string) bool {
+	return seq != 0 && endSeq != 0 && path != "" && grpcServer != ""
+}
+
+func main() {
+	exportMode := flag.String("export", "", "Set export mode: 'full' (export full ledger data of start_seq) or 'delta' (only export ledger diff data)")
+	seq := flag.Uint("start_seq", 0, "Starting sequence number")
+	endSeq := flag.Uint("end_seq", 0, "Ending sequence number")
+	path := flag.String("path", "", "Path to the data")
+	grpcServer := flag.String("grpc_server", "localhost:50051", "rippled's gRPC server address")
+	serverMode := flag.Bool("server", false, "Start server mode")
+	grpcPort := flag.Int("grpc_port", 0, "Port for gRPC server to listen on")
+	wsPort := flag.Int("ws_port", 0, "Port for WebSocket server to listen on")
+
+	flag.Parse()
```

Comment:
Better to create a structure of cli args. Also`flag` doesn't support short versions of cli options, maybe use `pflag` instead.

Reply:
We use a new library for parsing cli opts every tool we build :rage3:  let's settle on one library and use it? I quite liked the one we used in the DB cleaner.

Reply:
So `kingpin` it is ~